samples/ChangefeedMigrationSample/Program.cs

```diff
@@ -0,0 +1,222 @@
+        private string leaseCollectionName = "leaseCollectionName";
+
+        static void Main(string[] args)
+        {
+            Console.WriteLine("ChangeFeed App");
+            Program newApp = new ChangefeedMigrationSample.Program();
+            // Thread 1 comment out for thread 2 
+            newApp.RunChangeFeedProcessor();
+            // Thread 2 comment out for thread 1 
+            // UpdateDb(EndPointUrl, PrimaryKey); 
+            Console.ReadKey();
+        }
+
+        async void RunChangeFeedProcessor()
+        {
+            // connect client 
+            DocumentClient client = new DocumentClient(new Uri(uri), this.secretKey);
+            await client.CreateDatabaseIfNotExistsAsync(new Database { Id = this.dbName });
+
+            // create monitor collection if it does not exist 
+            // WARNING: CreateDocumentCollectionIfNotExistsAsync will create a new 
+            // with reserved through pul which has pricing implications. For details
+            // visit: https://azure.microsoft.com/en-us/pricing/details/cosmos-db/
+            await client.CreateDocumentCollectionIfNotExistsAsync(
+                UriFactory.CreateDatabaseUri(this.dbName),
+                new DocumentCollection { Id = this.monitoredCollectionName },
+                new RequestOptions { OfferThroughput = 400 });
+
+            // create lease collect if it does not exist
+            // WARNING: CreateDocumentCollectionIfNotExistsAsync will create a new 
+            // with reserved through pul which has pricing implications. For details
+            // visit: https://azure.microsoft.com/en-us/pricing/details/cosmos-db/
+            await client.CreateDocumentCollectionIfNotExistsAsync(
+            UriFactory.CreateDatabaseUri(this.dbName),
+            new DocumentCollection { Id = this.leaseCollectionName },
+            new RequestOptions { OfferThroughput = 400 });
+            await StartChangeFeedHost();
+        }
+
+        async Task StartChangeFeedHost()
+        {
+            string hostName = Guid.NewGuid().ToString();
+
+            // monitored collection info 
+            DocumentCollectionInfo documentCollectionLocation = new DocumentCollectionInfo
+            {
+                Uri = new Uri(this.uri),
+                MasterKey = this.secretKey,
+                DatabaseName = this.dbName,
+                CollectionName = this.monitoredCollectionName
+            };
+
+            // lease collection info 
+            DocumentCollectionInfo leaseCollectionLocation = new DocumentCollectionInfo
+            {
+                Uri = new Uri(this.uri),
+                MasterKey = this.secretKey,
+                DatabaseName = this.dbName,
+                CollectionName = this.leaseCollectionName
+            };
+
+            ChangeFeedEventHost host = new ChangeFeedEventHost(hostName, documentCollectionLocation, leaseCollectionLocation);
```

Comment:
Might want to expose an instance of ChangeFeedHostOptions as a parameter on the constructor to show that the leases intervals and polling can be customized.